clippy_lints/src/needless_pass_by_ref_mut.rs

```diff
@@ -46,20 +49,26 @@ declare_clippy_lint! {
     "using a `&mut` argument when it's not mutated"
 }
 
-#[derive(Copy, Clone)]
-pub struct NeedlessPassByRefMut {
+#[derive(Clone)]
+pub struct NeedlessPassByRefMut<'tcx> {
     avoid_breaking_exported_api: bool,
+    used_fn_def_ids: FxHashSet<LocalDefId>,
+    // TODO(Centri3): Change this to a `Vec`. This will cause a performance deficit but it'll make the output a lot
+    // more consistent, as adding or removing a function won't change the order of them
+    fn_def_ids_to_maybe_unused_mut: FxHashMap<LocalDefId, Vec<rustc_hir::Ty<'tcx>>>,
```

Comment:
No need to use a `Vec`, simply replace `FxHashMap` with `FxIndexMap` to keep the order.

Reply:
Neat, had no idea that was a crate

Reply:
Compiler internals are full of very useful things. :D